extensions/shared/use-upgrade-flow/index.js

```diff
@@ -24,10 +24,18 @@ function redirect( url, callback ) {
 	window.top.location.href = url;
 }
 
+function isCheckoutOverlayAvailable() {
+	try {
+		return window.wp.hooks.hasAction( 'a8c.wpcom-block-editor.openCheckoutModal' );
+	} catch ( err ) {
+		return false;
+	}
```

Comment:
Hmm perhaps the `try...catch` is not necessary, since `hasAction()` would return `true` or `false`?

Reply:
`wp.hooks` is available in wp-admin right? Its WP core not Calypso specific so I think it's there all the time.

Reply:
In theory yes it should always be there.

This was just a safety feature in case this code is run in a situation where any of the properties on the window object do not exist.

Then it will return false and fallback to the redirect method.

Reply:
Hmm, I was doing some digging for.. prior art...and I'm surprised that in jetpack this is the only part that is calling `wp.hooks`!

Reply:
> I'm surprised that in jetpack this is the only part that is calling `wp.hooks`!

We do rely on the Hooks package in other parts of the codebase. Check for `@wordpress/hooks` imports in the `extensions/` directory.

Reply:
Thanks @jeherve, does that mean we can replace:

`window.wp.hooks.hasAction( 'a8c.wpcom-block-editor.openCheckoutModal' );`

with:

```
import { hasAction } from '@wordpress/hooks';
// ...

window.wp.hooks.hasAction( 'a8c.wpcom-block-editor.openCheckoutModal' );
```

Reply:
I haven't checked this branch out yet, but in general yes, that would be the preferred approach.

Reply:
Thanks for pointing us to the right direction! :) Turns out I was using the wrong search keyword! Cool thanks!